File: crypt.py
import base64
import hashlib
from Crypto.Cipher import AES
import json


def pad(data):
    block_size = AES.block_size
    padding = block_size - len(data) % block_size
    return data + bytes([padding] * padding)


def encrypt(data, passphrase):
    secret_key = hashlib.sha256(passphrase.encode()).digest()
    iv = AES.new(secret_key, AES.MODE_CBC).iv
    cipher = AES.new(secret_key, AES.MODE_CBC, iv)
    data = data.encode('utf-8')
    padded_data = pad(data)   
    encrypted_data = cipher.encrypt(padded_data)
    iv_64 = base64.b64encode(iv).decode()
    encrypted_64 = base64.b64encode(encrypted_data).decode()
    json_data = {"iv": iv_64, "data": encrypted_64}
    return base64.b64encode(json.dumps(json_data).encode()).decode()


def decrypt(encrypted_data, passphrase):
    secret_key = hashlib.sha256(passphrase.encode()).digest()
    json_data = json.loads(base64.b64decode(encrypted_data))
    iv = base64.b64decode(json_data['iv'])
    encrypted_data = base64.b64decode(json_data['data'])
    cipher = AES.new(secret_key, AES.MODE_CBC, iv)
    decrypted_data = cipher.decrypt(encrypted_data) 
    decrypted_data = decrypted_data[:-decrypted_data[-1]]
    return decrypted_data.decode('utf-8')

File: crypt.php
<?php
function encrypt($data, $passphrase)
{
    $secret_key = hash('sha256', $passphrase, true);
    $iv = openssl_random_pseudo_bytes(openssl_cipher_iv_length('aes-256-cbc'));
    $encrypted_64 = openssl_encrypt($data, 'aes-256-cbc', $secret_key, 0, $iv);
    $iv_64 = base64_encode($iv);
    $json = new stdClass();
    $json->iv = $iv_64;
    $json->data = $encrypted_64;
    return base64_encode(json_encode($json));
}

function decrypt($data, $passphrase)
{
    $secret_key = hash('sha256', $passphrase, true);
    $json = json_decode(base64_decode($data));
    $iv = base64_decode($json->{'iv'});
    $encrypted_64 = $json->{'data'};
    $data_encrypted = base64_decode($encrypted_64);
    $decrypted = openssl_decrypt($data_encrypted, 'aes-256-cbc', $secret_key, OPENSSL_RAW_DATA, $iv);
    return $decrypted;
}
